contracts/vault/src/execute.rs

```diff
@@ -141,88 +144,41 @@ fn execute_redistribute_funds(deps: DepsMut, env: Env, info: MessageInfo) -> Res
-                .query(&QueryRequest::Wasm(WasmQuery::Smart {
-                    contract_addr: market_id.clone(),
-                    msg: to_json_binary(&MarketQueryMsg::Status { price: None }).ok()?,
-                }))
-                .ok()?;
-            if let Token::Native { denom, .. } = &resp.collateral {
-                if denom == &config.usdc_denom {
-                    Some(Ok(market_id))
-                } else {
-                    None
-                }
-            } else {
-                None
-            }
-        })
-        .collect::<StdResult<Vec<String>>>()?;
-
-    let mut utilizations: Vec<(String, Uint128)> = markets
-        .into_iter()
-        .filter_map(|market| {
-            let resp: StatusResp = deps
-                .querier
-                .query(&QueryRequest::Wasm(WasmQuery::Smart {
-                    contract_addr: market.clone(),
-                    msg: to_json_binary(&MarketQueryMsg::Status { price: None })
-                        .expect("Serialize Market Query Msg"),
-                }))
-                .ok()?;
-
-            let utilization =
-                (resp.liquidity.total_lp + resp.liquidity.total_xlp).unwrap_or_default();
-
-            let value = Uint128::from(utilization.into_u128().expect("Error LpToken to Uint128"));
-            Some((market, value))
-        })
-        .collect();
-
-    utilizations.sort_by(|a, b| b.1.cmp(&a.1));
-
-    let total_bps: u16 = config.markets_allocation_bps.iter().sum();
+    let total_bps: u16 = config.markets_allocation_bps.values().sum();
     if total_bps == 0 {
         return Err(anyhow!("No allocation percentages defined"));
     }
 
     let mut messages: Vec<CosmosMsg> = vec![];
     let mut remaining = excess;
 
-    for (i, bps) in config.markets_allocation_bps.iter().enumerate() {
-        if let Some((market, _)) = utilizations.get(i) {
-            let amount = excess.multiply_ratio(*bps as u128, total_bps as u128);
-            if !amount.is_zero() {
-                let deposit_msg = WasmMsg::Execute {
-                    contract_addr: market.clone(),
-                    msg: to_json_binary(&MarketExecuteMsg::DepositLiquidity {
-                        stake_to_xlp: false,
-                    })?,
-                    funds: vec![Coin {
-                        denom: config.usdc_denom.clone(),
-                        amount,
-                    }],
-                };
-                messages.push(deposit_msg.into());
-
-                state::MARKET_ALLOCATIONS.update(
-                    deps.storage,
-                    market.as_str(),
-                    |a| -> Result<Uint128, StdError> { Ok(a.unwrap_or(Uint128::zero()) + amount) },
-                )?;
-
-                remaining = remaining.saturating_sub(amount);
-            }
+    for (market, allocation_bps) in config.markets_allocation_bps.iter() {
+        let amount = excess.multiply_ratio(*allocation_bps as u128, total_bps as u128);
```

Comment:
I'd prefer to use `.into()` for upcasting to a `u128` instead. Motivation: as-casting can panic/truncate when moving to smaller data types, so we try to avoid using it overall.